src/main/java/edu/berkeley/ground/dao/usage/cassandra/CassandraLineageGraphVersionFactory.java

```diff
@@ -148,15 +149,7 @@ public LineageGraphVersion retrieveFromDatabase(long id) throws GroundException
 
     long lineageGraphId = resultSet.getLong("lineage_graph_id");
 
-    List<Long> lineageEdgeVersionIds = new ArrayList<>();
-    CassandraResults lineageEdgeSet = this.dbClient.equalitySelect("lineage_graph_version_edge",
-        DbClient.SELECT_STAR, lineageEdgePredicate);
-
-    if (!lineageEdgeSet.isEmpty()) {
-      do {
-        lineageEdgeVersionIds.add(lineageEdgeSet.getLong("lineage_edge_version_id"));
-      } while (lineageEdgeSet.next());
-    }
+    List<Long> lineageEdgeVersionIds = resultSet.getSet("lineage_edge_version_id_set", Long.class).stream().collect(Collectors.toList());
```

Comment:
Can't you create a new `List` from a `Set` instead of streaming? See [this constructor](https://docs.oracle.com/javase/8/docs/api/java/util/ArrayList.html#ArrayList-java.util.Collection-).